libs/libc/semaphore/sem_init.c

```diff
@@ -62,15 +62,15 @@
  *
  ****************************************************************************/
 
-int nxsem_init(FAR sem_t *sem, int pshared, unsigned int value)
+int nxsem_init(FAR sem_t *sem, int pshared, int32_t value)
```

Comment:
Is there some reason to change the prototype ? The prototype of sem_init is

int sem_init(sem_t *sem, int pshared, unsigned int value);

Reply:
I now see why you had to do it:
 Error: misc/lib_mutex.c:94:40: error: large integer implicitly truncated to unsigned type [-Werror=overflow]
   int ret = nxsem_init(&mutex->sem, 0, NXSEM_NO_MHOLDER);

On some platforms unsigned int is 16-bit, not large enough to hold the u32 mask for NXSEM_NO_MHOLDER

Reply:
Yes, and I had already forgotten this; I believe changing it is fine since it is anyhow nuttx internal interface, and initializing a 32-bit value (both counter and holder are 32 bits nowdays). But, it probably should be uint32_t; it would make no sense to initialize a counting semaphore with a negative value...